### PR TITLE
DSD-1439: Storybook 7 Icon documentation

### PR DESCRIPTION
**Pull request by EdwinGuzman**
Fixes JIRA ticket [DSD-1439](https://jira.nypl.org/browse/DSD-1439)

## This PR does the following:

- Updating Storybook documentation for the `Icon` component.
- Consolidated some of the component array variables into a single file.
- Compare with prod: https://nypl.github.io/nypl-design-system/reservoir/v1/?path=/docs/components-media-icons-icon--icon-with-controls

## How has this been tested?

Locally.

## Accessibility concerns or updates

- N/A

### Checklist:

- [ ] I have updated the Storybook documentation accordingly.
- [ ] I have added relevant accessibility documentation for this pull request.
- [ ] All new and existing tests passed.

### Front End Review:

- [ ] Review the Vercel preview deployment once it is ready.

### CODE REVIEW

File: src/components/Icons/iconVariables.ts [hunk 1]
```diff
@@ -0,0 +1,110 @@
+export const iconAlignArray = ["left", "right", "none"] as const;
```
Comment (bigfishdesign13) [inline src/components/Icons/iconVariables.ts:1]: I've noticed that you now are using `as const` on some export statements.  Why is that?

File: src/components/Icons/Icon.stories.tsx [hunk 2]
```diff
@@ -0,0 +1,222 @@
...
+  iconSizesArray,
+} from "./iconVariables";
+
+const meta: Meta<typeof Icon> = {
+  title: "Components/Media & Icons/Icon",
+  component: Icon,
+  decorators: [withDesign],
+  argTypes: {
+    align: {
+      control: { type: "select" },
```
Comment (bigfishdesign13) [inline src/components/Icons/Icon.stories.tsx]: Change to `type: "radio"`.

File: src/components/Icons/Icon.stories.tsx [hunk 3]
```diff
@@ -0,0 +1,222 @@
...
+    },
+    children: { table: { disable: true } },
+    className: { control: false },
+    color: {
+      control: { type: "select" },
+      options: iconColorsArray,
+      table: { defaultValue: { summary: "ui.black" } },
+    },
+    iconRotation: {
+      control: { type: "select" },
```
Comment (bigfishdesign13) [inline src/components/Icons/Icon.stories.tsx]: Change to `type: "radio"`.

**Review by bigfishdesign13 — CHANGES_REQUESTED** [review]
Some minor changes.

File: src/components/Icons/iconVariables.ts (hunk 1 above)
Comment (EdwinGuzman) [reply src/components/Icons/iconVariables.ts:1]: It's not related to the export statement. `const iconAlignArray = ["left", "right", "none"] as const;` creates the array as read-only so we have a defined set of values. This makes it easier for typescript to create a type from the values. But the array is helpful to define the options in the storybook doc config.

**Review by bigfishdesign13 — APPROVED** [review]
💯

**Pull request merged**